ampcrowd/basecrowd/models.py

```diff
@@ -150,3 +171,43 @@ def add_model_rels(self):
         # responses pertain to a task
         self.add_rel(self.response_model, self.task_model, models.ForeignKey,
                      'task', 'responses')
+
+
+class TemplateResource(models.Model):
+    name = models.CharField(max_length=200)
+    content = models.TextField()
+    direct_dependencies = models.ManyToManyField('self', symmetrical=False, related_name="upstream_dependencies")
+    direct_requirements = models.ManyToManyField('self', symmetrical=False, related_name="upstream_requirements")
```

Comment:
I'm still not totally understanding. Why would an import not also be a requirement? Are you trying to support the use case where templates gracefully degrade if some (non-required) imports are not available? Or are you trying to differentiate between template inheritance and including external libraries?